// my-chai-ui/index.js
function applyChaiStyles() {

  const theme = {
    primary: '#2563eb',
    secondary: '#f5f5f5',
    text: '#111'
  };

  document.querySelectorAll('*').forEach(el => {

    el.classList.forEach(cls => {

      if (!cls.startsWith('chai-')) return;

      const [_, type, value] = cls.split('-');

      /* =======================
         🔹 UTILITIES
      ======================= */

      if (type === 'p') el.style.padding = value + 'px';
      if (type === 'm') el.style.margin = value + 'px';
      if (type === 'bg') el.style.backgroundColor = value;

      if (type === 'text') {
        if (value === 'center') el.style.textAlign = 'center';
        else el.style.color = value;
      }

      if (type === 'fs') el.style.fontSize = value + 'px';

      if (type === 'border') el.style.border = value + 'px solid black';
      if (type === 'rounded') el.style.borderRadius = value + 'px';

      if (type === 'flex') el.style.display = 'flex';
      if (type === 'items' && value === 'center') el.style.alignItems = 'center';
      if (type === 'justify' && value === 'center') el.style.justifyContent = 'center';

      if (type === 'gap') el.style.gap = value + 'px';

      /* =======================
         🔹 COMPONENTS
      ======================= */

      if (cls === 'chai-card') {
        el.style.padding = '20px';
        el.style.backgroundColor = theme.secondary;
        el.style.color = theme.text;
        el.style.borderRadius = '12px';
        el.style.boxShadow = '0 4px 10px rgba(0,0,0,0.1)';
      }

      if (cls === 'chai-btn') {
        el.style.padding = '12px 24px';
        el.style.backgroundColor = theme.primary;
        el.style.color = '#fff';
        el.style.border = 'none';
        el.style.borderRadius = '8px';
        el.style.cursor = 'pointer';
      }

      if (cls === 'chai-btn-danger') el.style.backgroundColor = 'red';

      if (cls === 'chai-avatar') {
        el.style.width = '60px';
        el.style.height = '60px';
        el.style.borderRadius = '50%';
        el.style.objectFit = 'cover';
      }

      if (cls === 'chai-progress') {
        el.style.width = '100%';
        el.style.height = '10px';
        el.style.backgroundColor = '#eee';
        el.style.borderRadius = '10px';
        el.style.overflow = 'hidden';

        const bar = document.createElement('div');
        bar.style.height = '100%';
        bar.style.width = '0%';
        bar.style.backgroundColor = theme.primary;
        bar.style.transition = 'width 1s';

        el.appendChild(bar);
      }

      /* =======================
         🔹 TABLE
      ======================= */

      if (cls === 'chai-table') {
        el.style.width = '100%';
        el.style.borderCollapse = 'collapse';

        el.querySelectorAll('th, td').forEach(cell => {
          cell.style.border = '1px solid #ddd';
          cell.style.padding = '10px';
        });

        el.querySelectorAll('th').forEach(th => {
          th.style.backgroundColor = '#f5f5f5';
        });
      }

      /* =======================
         🔹 DROPDOWN
      ======================= */

      if (cls === 'chai-dropdown') el.style.position = 'relative';

      if (cls === 'chai-dropdown-menu') {
        el.style.position = 'absolute';
        el.style.top = '100%';
        el.style.left = '0';
        el.style.display = 'none';
        el.style.background = '#fff';
        el.style.border = '1px solid #ddd';
        el.style.borderRadius = '8px';
      }

      /* =======================
         🔹 MODAL
      ======================= */

      if (cls === 'chai-modal') {
        el.style.position = 'fixed';
        el.style.top = '0';
        el.style.left = '0';
        el.style.width = '100%';
        el.style.height = '100%';
        el.style.background = 'rgba(0,0,0,0.5)';
        el.style.display = 'none';
        el.style.justifyContent = 'center';
        el.style.alignItems = 'center';
      }

      if (cls === 'chai-modal-content') {
        el.style.background = '#fff';
        el.style.padding = '20px';
        el.style.borderRadius = '10px';
      }

      /* =======================
         🔹 HOVER
      ======================= */

      if (cls === 'chai-hover-scale') {
        el.addEventListener('mouseenter', () => el.style.transform = 'scale(1.05)');
        el.addEventListener('mouseleave', () => el.style.transform = 'scale(1)');
      }

      /* =======================
         🔹 ANIMATION
      ======================= */

      if (cls === 'chai-animate-fade') {
        el.style.opacity = '0';
        el.style.transition = '0.5s';
        setTimeout(() => el.style.opacity = '1', 100);
      }

      el.classList.remove(cls);

    });

  });

}

/* =======================
   🔹 HELPERS
======================= */

function openModal() {
  document.getElementById('myModal').style.display = 'flex';
}

function closeModal() {
  document.getElementById('myModal').style.display = 'none';
}

function showToast(msg) {
  const t = document.createElement('div');
  t.innerText = msg;

  Object.assign(t.style, {
    position: 'fixed',
    bottom: '20px',
    right: '20px',
    background: '#333',
    color: '#fff',
    padding: '10px 20px',
    borderRadius: '8px'
  });

  document.body.appendChild(t);
  setTimeout(() => t.remove(), 3000);
}

function setProgress(el, value) {
  const bar = el.firstChild;
  if (bar) bar.style.width = value + '%';
}

/* =======================
   🔹 EVENTS
======================= */

document.addEventListener("DOMContentLoaded", () => {
  applyChaiStyles();

  document.querySelectorAll('.chai-dropdown').forEach(drop => {
    const btn = drop.querySelector('button');
    const menu = drop.querySelector('.chai-dropdown-menu');

    btn?.addEventListener('click', () => {
      menu.style.display = menu.style.display === 'block' ? 'none' : 'block';
    });
  });
});


if (typeof window !== "undefined") {
  window.applyChaiStyles = applyChaiStyles;
}

export { applyChaiStyles };
